src/modules/order/queues/order.processor.ts

```diff
@@ -0,0 +1,126 @@
+import { UserRole } from 'src/modules/user/constants/user.enum';
+import { USER_METADATA_KEY } from 'src/modules/user/constants/user-metadata.constant';
+import { Order } from '../entities/order.entity';
+import { Shipper } from 'src/modules/shipper/entities/shipper.entity';
+import { User } from 'src/modules/user/entities/user.entity';
+import { RedisClient } from 'nestjs-redis/dist/redis-client.provider';
+import { Redis } from 'ioredis';
+import { ORDER_STATUS } from '../constants/order-status.constant';
+
+@Processor(QueueName.ORDER, {
+  concurrency: 1,
+})
+export class OrderProcessor extends WorkerHost {
+  private logger = new Logger(OrderProcessor.name);
+  private redisClient = new Redis();
+
+  constructor(
+    private readonly orderService: OrderService,
+    private readonly userService: UserService,
+    private readonly userMetadataService: UserMetadataService,
+    @InjectQueue(QueueName.ORDER)
+    private readonly orderQueue: Queue,
+  ) {
+    super();
+  }
+
+  async process(job: Job<any, any, string>, token?: string): Promise<any> {
+    const order = job.data as Order;
+    try {
+      switch (job.name) {
+        case 'order-find-shipper':
+          return await this.handleFindShipper(order);
+        default:
+          throw new Error('No job name match');
+      }
+    } catch (error) {
+      await this.handleJobError(job, error);
+      await this.processFailedJob(job, error);
+      this.logger.error(error);
+    }
+  }
+
+  //Như này không phải cách, Code này đã bị lặp lại 2 lần ở shop processor (không biết nên đặt đâu để cả 2 processor đặt được)
+  async processFailedJob(job: Job, error: HttpException) {
+    if (error instanceof HttpException) {
+      Sentry.captureException(error, {
+        tags: {
+          jobId: job.id,
+          jobName: job.name,
+        },
+        extra: {
+          jobId: job.id,
+          jobName: job.name,
+          jobData: job.data,
+          jobPayloadError: error,
+        },
+      });
+    }
+  }
+
+  // find shipper
+  async handleFindShipper(order: Order) {
+    const foundShippers = await this.userService.findAllBy({
+      role: UserRole.SHIPPER,
+      userMetadata: { key: USER_METADATA_KEY.SHIPPER_STATUS, value: 'ready' },
+    });
+    console.log('Check shipper:: ', foundShippers);
+
+    if (!foundShippers || foundShippers.length === 0) {
+      throw new BadRequestException('Shipper not ready :)');
+    }
+
+    await this.handleFindShipperSuccess(foundShippers, order);
+  }
+
+  async handleFindShipperSuccess(shippers: User[], order: Order) {
+    for (const shipper of shippers) {
+      const updatedOrder = await this.orderService.update(order.id, {
+        shipperId: shipper.id,
+      });
+
+      await this.orderService.updateShipperStatus(shipper.id, 'shipping');
+
+      await this.redisClient.del(this.makeJobKey(order.id));
+      return updatedOrder;
+    }
+  }
+
+  async handleJobError(job: Job, error: Error): Promise<void> {
```

Comment:
Oops! i was thought you ask me if the order couldn't find shipper for them then cause error :v
It's ok but you can using https://docs.nestjs.com/techniques/queues#job-options to make this attempts btw